### PR TITLE
feat: add table of contents

### LINKED ISSUE #737: feat: add individual TOC for each docs page

### PR DESCRIPTION
**Pull request by StephenFluin**
Also standardized on in-content H1s away from pagetitles 
Fixed missing H2s on gmp overview

Fixes #737

### CODE REVIEW

**Comment by StephenFluin** [comment]
Asking for more eyes on this one @Olanetsoft @benjamin852 to make sure I didn't miss any edge cases.

File: src/layouts/TableOfContents.astro [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  return toc;
+}
+---
+
+<div id="toc">
+  <strong>On this page</strong>
+  <ul>
+    {toc.map((heading) => <TableOfContentsHeading heading={heading} />)}
+  </ul>
+  <!-- <svg class="toc-progress" xmlns="http://www.w3.org/2000/svg">
```
Comment (benjamin852) [inline src/layouts/TableOfContents.astro:36]: can we remove this commented code?

**Comment by benjamin852** [comment]
Table of contents looks good to me! Two things though
1. If we have a page without any headers for the TOC do we want to just remove it? Or perhaps just put the title of the page under the toc. Because right now for pages such as `/node/config-cli` the TOC just say "on this page" and then nothing underneath it.
2. /validator/setup/health-check is 404ing

**Comment by StephenFluin** [comment]
> Table of contents looks good to me! Two things though
> 
> 1. If we have a page without any headers for the TOC do we want to just remove it? Or perhaps just put the title of the page under the toc. Because right now for pages such as `/node/config-cli` the TOC just say "on this page" and then nothing underneath it.
> 2. /validator/setup/health-check is 404ing

1. Modified so we only show if there are 3 or more sections
2. I think that's just the auto-forwards not working in the preview, the link is wrong globally. I'll send another PR

**Review by benjamin852 — APPROVED** [review]

**Pull request merged**